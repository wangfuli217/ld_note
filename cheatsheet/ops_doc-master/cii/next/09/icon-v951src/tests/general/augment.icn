record array(a,b,c,d,e,f,g)

procedure p1()
   write("i := 10 ----> ",image(i := 10) | "none")
   write("i =:= 9 ----> ",image(i =:= 9) | "none")
   write("i ----> ",image(i) | "none")
   write("i := 10 ----> ",image(i := 10) | "none")
   write("i =:= 10 ----> ",image(i =:= 10) | "none")
   write("i ----> ",image(i) | "none")
   write("i := 10 ----> ",image(i := 10) | "none")
   write("i =:= 11 ----> ",image(i =:= 11) | "none")
   write("i ----> ",image(i) | "none")
   write("i := 10 ----> ",image(i := 10) | "none")
   write("i >=:= 9 ----> ",image(i >=:= 9) | "none")
end

procedure p2()
   write("i ----> ",image(i) | "none")
   write("i := 10 ----> ",image(i := 10) | "none")
   write("i >=:= 10 ----> ",image(i >=:= 10) | "none")
   write("i ----> ",image(i) | "none")
   write("i := 10 ----> ",image(i := 10) | "none")
   write("i >=:= 11 ----> ",image(i >=:= 11) | "none")
   write("i ----> ",image(i) | "none")
   write("i := 10 ----> ",image(i := 10) | "none")
   write("i >:= 9 ----> ",image(i >:= 9) | "none")
   write("i ----> ",image(i) | "none")
   write("i := 10 ----> ",image(i := 10) | "none")
end

procedure p3()
   write("i >:= 10 ----> ",image(i >:= 10) | "none")
   write("i ----> ",image(i) | "none")
   write("i := 10 ----> ",image(i := 10) | "none")
   write("i >:= 11 ----> ",image(i >:= 11) | "none")
   write("i ----> ",image(i) | "none")
   write("i := 10 ----> ",image(i := 10) | "none")
   write("i <=:= 9 ----> ",image(i <=:= 9) | "none")
   write("i ----> ",image(i) | "none")
   write("i := 10 ----> ",image(i := 10) | "none")
   write("i <=:= 10 ----> ",image(i <=:= 10) | "none")
   write("i ----> ",image(i) | "none")
end

procedure p4()
   write("i := 10 ----> ",image(i := 10) | "none")
   write("i <=:= 11 ----> ",image(i <=:= 11) | "none")
   write("i ----> ",image(i) | "none")
   write("i := 10 ----> ",image(i := 10) | "none")
   write("i <:= 9 ----> ",image(i <:= 9) | "none")
   write("i ----> ",image(i) | "none")
   write("i := 10 ----> ",image(i := 10) | "none")
   write("i <:= 10 ----> ",image(i <:= 10) | "none")
   write("i ----> ",image(i) | "none")
   write("i := 10 ----> ",image(i := 10) | "none")
   write("i <:= 11 ----> ",image(i <:= 11) | "none")
end

procedure p5()
   write("i ----> ",image(i) | "none")
   write("i := 10 ----> ",image(i := 10) | "none")
   write("i ~=:= 9 ----> ",image(i ~=:= 9) | "none")
   write("i ----> ",image(i) | "none")
   write("i := 10 ----> ",image(i := 10) | "none")
   write("i ~=:= 10 ----> ",image(i ~=:= 10) | "none")
   write("i ----> ",image(i) | "none")
   write("i := 10 ----> ",image(i := 10) | "none")
   write("i ~=:= 11 ----> ",image(i ~=:= 11) | "none")
   write("i ----> ",image(i) | "none")
   write("i := 10 ----> ",image(i := 10) | "none")
end

procedure p6()
   write("i +:= 9 ----> ",image(i +:= 9) | "none")
   write("i ----> ",image(i) | "none")
   write("i := 10 ----> ",image(i := 10) | "none")
   write("i +:= 10 ----> ",image(i +:= 10) | "none")
   write("i ----> ",image(i) | "none")
   write("i := 10 ----> ",image(i := 10) | "none")
   write("i +:= 11 ----> ",image(i +:= 11) | "none")
   write("i ----> ",image(i) | "none")
   write("i := 10 ----> ",image(i := 10) | "none")
   write("i -:= 9 ----> ",image(i -:= 9) | "none")
   write("i ----> ",image(i) | "none")
end

procedure p7()
   write("i := 10 ----> ",image(i := 10) | "none")
   write("i -:= 10 ----> ",image(i -:= 10) | "none")
   write("i ----> ",image(i) | "none")
   write("i := 10 ----> ",image(i := 10) | "none")
   write("i -:= 11 ----> ",image(i -:= 11) | "none")
   write("i ----> ",image(i) | "none")
   write("i := 10 ----> ",image(i := 10) | "none")
   write("i *:= 9 ----> ",image(i *:= 9) | "none")
   write("i ----> ",image(i) | "none")
   write("i := 10 ----> ",image(i := 10) | "none")
   write("i *:= 10 ----> ",image(i *:= 10) | "none")
end

procedure p8()
   write("i ----> ",image(i) | "none")
   write("i := 10 ----> ",image(i := 10) | "none")
   write("i *:= 11 ----> ",image(i *:= 11) | "none")
   write("i ----> ",image(i) | "none")
   write("i := 10 ----> ",image(i := 10) | "none")
   write("i /:= 9 ----> ",image(i /:= 9) | "none")
   write("i ----> ",image(i) | "none")
   write("i := 10 ----> ",image(i := 10) | "none")
   write("i /:= 10 ----> ",image(i /:= 10) | "none")
   write("i ----> ",image(i) | "none")
   write("i := 10 ----> ",image(i := 10) | "none")
end

procedure p9()
   write("i /:= 11 ----> ",image(i /:= 11) | "none")
   write("i ----> ",image(i) | "none")
   write("i := 10 ----> ",image(i := 10) | "none")
   write("i %:= 9 ----> ",image(i %:= 9) | "none")
   write("i ----> ",image(i) | "none")
   write("i := 10 ----> ",image(i := 10) | "none")
   write("i %:= 10 ----> ",image(i %:= 10) | "none")
   write("i ----> ",image(i) | "none")
   write("i := 10 ----> ",image(i := 10) | "none")
   write("i %:= 11 ----> ",image(i %:= 11) | "none")
   write("i ----> ",image(i) | "none")
end

procedure p10()
   write("i := 10 ----> ",image(i := 10) | "none")
   write("i ^:= 9 ----> ",image(i ^:= 9) | "none")
   write("i ----> ",image(i) | "none")
   write("i := 10 ----> ",image(i := 10) | "none")
   write("s := \"x\" ----> ",image(s := "x") | "none")
   write("s <<:= \"x\" ----> ",image(s <<:= "x") | "none")
end

procedure p11()
   write("s ----> ",image(s) | "none")
   write("s := \"x\" ----> ",image(s := "x") | "none")
   write("s <<:= \"xx\" ----> ",image(s <<:= "xx") | "none")
   write("s ----> ",image(s) | "none")
   write("s := \"x\" ----> ",image(s := "x") | "none")
   write("s <<:= \"X\" ----> ",image(s <<:= "X") | "none")
   write("s ----> ",image(s) | "none")
   write("s := \"x\" ----> ",image(s := "x") | "none")
   write("s <<:= \"abc\" ----> ",image(s <<:= "abc") | "none")
   write("s ----> ",image(s) | "none")
   write("s := \"x\" ----> ",image(s := "x") | "none")
end

procedure p12()
   write("s ~==:= \"x\" ----> ",image(s ~==:= "x") | "none")
   write("s ----> ",image(s) | "none")
   write("s := \"x\" ----> ",image(s := "x") | "none")
   write("s ~==:= \"xx\" ----> ",image(s ~==:= "xx") | "none")
   write("s ----> ",image(s) | "none")
   write("s := \"x\" ----> ",image(s := "x") | "none")
   write("s ~==:= \"X\" ----> ",image(s ~==:= "X") | "none")
   write("s ----> ",image(s) | "none")
   write("s := \"x\" ----> ",image(s := "x") | "none")
   write("s ~==:= \"abc\" ----> ",image(s ~==:= "abc") | "none")
   write("s ----> ",image(s) | "none")
end

procedure main()
   p1()
   p2()
   p3()
   p4()
   p5()
   p6()
   p7()
   p8()
   p9()
   p10()
   p11()
   p12()
end

global i, s, c, one, two, x
