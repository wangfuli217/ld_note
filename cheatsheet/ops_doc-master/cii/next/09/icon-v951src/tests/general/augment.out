i := 10 ----> 10
i =:= 9 ----> none
i ----> 10
i := 10 ----> 10
i =:= 10 ----> 10
i ----> 10
i := 10 ----> 10
i =:= 11 ----> none
i ----> 10
i := 10 ----> 10
i >=:= 9 ----> 9
i ----> 9
i := 10 ----> 10
i >=:= 10 ----> 10
i ----> 10
i := 10 ----> 10
i >=:= 11 ----> none
i ----> 10
i := 10 ----> 10
i >:= 9 ----> 9
i ----> 9
i := 10 ----> 10
i >:= 10 ----> none
i ----> 10
i := 10 ----> 10
i >:= 11 ----> none
i ----> 10
i := 10 ----> 10
i <=:= 9 ----> none
i ----> 10
i := 10 ----> 10
i <=:= 10 ----> 10
i ----> 10
i := 10 ----> 10
i <=:= 11 ----> 11
i ----> 11
i := 10 ----> 10
i <:= 9 ----> none
i ----> 10
i := 10 ----> 10
i <:= 10 ----> none
i ----> 10
i := 10 ----> 10
i <:= 11 ----> 11
i ----> 11
i := 10 ----> 10
i ~=:= 9 ----> 9
i ----> 9
i := 10 ----> 10
i ~=:= 10 ----> none
i ----> 10
i := 10 ----> 10
i ~=:= 11 ----> 11
i ----> 11
i := 10 ----> 10
i +:= 9 ----> 19
i ----> 19
i := 10 ----> 10
i +:= 10 ----> 20
i ----> 20
i := 10 ----> 10
i +:= 11 ----> 21
i ----> 21
i := 10 ----> 10
i -:= 9 ----> 1
i ----> 1
i := 10 ----> 10
i -:= 10 ----> 0
i ----> 0
i := 10 ----> 10
i -:= 11 ----> -1
i ----> -1
i := 10 ----> 10
i *:= 9 ----> 90
i ----> 90
i := 10 ----> 10
i *:= 10 ----> 100
i ----> 100
i := 10 ----> 10
i *:= 11 ----> 110
i ----> 110
i := 10 ----> 10
i /:= 9 ----> 1
i ----> 1
i := 10 ----> 10
i /:= 10 ----> 1
i ----> 1
i := 10 ----> 10
i /:= 11 ----> 0
i ----> 0
i := 10 ----> 10
i %:= 9 ----> 1
i ----> 1
i := 10 ----> 10
i %:= 10 ----> 0
i ----> 0
i := 10 ----> 10
i %:= 11 ----> 10
i ----> 10
i := 10 ----> 10
i ^:= 9 ----> 1000000000
i ----> 1000000000
i := 10 ----> 10
s := "x" ----> "x"
s <<:= "x" ----> none
s ----> "x"
s := "x" ----> "x"
s <<:= "xx" ----> "xx"
s ----> "xx"
s := "x" ----> "x"
s <<:= "X" ----> none
s ----> "x"
s := "x" ----> "x"
s <<:= "abc" ----> none
s ----> "x"
s := "x" ----> "x"
s ~==:= "x" ----> none
s ----> "x"
s := "x" ----> "x"
s ~==:= "xx" ----> "xx"
s ----> "xx"
s := "x" ----> "x"
s ~==:= "X" ----> "X"
s ----> "X"
s := "x" ----> "x"
s ~==:= "abc" ----> "abc"
s ----> "abc"
